### PR TITLE
Should not reconnect, since it mess up transactions

### PR DESCRIPTION
**Pull request by jhthorsen**

### CODE REVIEW

File: lib/Mojo/Redis/Database.pm [hunk 1]
```diff
@@ -121,14 +121,20 @@ sub _add_method {
   }
 }
 
+sub _connection {
+  my ($self, $non_blocking) = @_;
+  $self->connection($self->redis->_blocking_connection) if !$non_blocking and !$self->{connection};
```
Comment (Grinnz) [inline lib/Mojo/Redis/Database.pm]: Unsure about this; isn't the connection attribute used for non-blocking connections? Would storing the blocking connection there break later non-blocking queries from the same $db?

Comment (jhthorsen) [reply lib/Mojo/Redis/Database.pm]: I need to store the blocking connection on the database object, so the next blocking call reuse the same connection.

I considered adding a check so you cannot go from blocking to non-blocking and the other way around. I don't think I want a single $db to be able to do both.

Comment (jhthorsen) [reply lib/Mojo/Redis/Database.pm]: How about if I add the check and die if the $db is used in both context, so don't get into a "deadlock" situation?

File: lib/Mojo/Redis/Database.pm [hunk 2]
```diff
@@ -121,14 +120,24 @@ sub _add_method {
...
 }
 
+sub connection {
+  my $self = shift;
+
+  # Back compat: $self->connection(Mojo::Redis::Connection->new);
+  $self->{_conn_dequeue} = shift if UNIVERSAL::isa($_[0], 'Mojo::Redis::Connection');
+
+  my $method = $_[0] ? '_blocking_connection' : '_dequeue';
+  return $self->{"_conn$method"} ||= $self->redis->$method;
```
[the remark this answers is not in the document]
Comment (Grinnz) [reply lib/Mojo/Redis/Database.pm]: :+1:

`UNIVERSAL::isa($_[0], 'Mojo::Redis::Connection')` should be `blessed $_[0] && $_[0]->isa('Mojo::Redis::Connection')` though.

[the remark this answers is not in the document]
Comment (jhthorsen) [reply lib/Mojo/Redis/Database.pm]: I never understood what people got against UNIVERSAL::isa() 😞

[the remark this answers is not in the document]
Comment (Grinnz) [reply lib/Mojo/Redis/Database.pm]: It does not respect an overriden or inherited `isa` method.

**Pull request merged**